pkg/controller/buildrun/generate_taskrun_test.go

```diff
@@ -76,8 +88,9 @@ func TestGenerateTaskSpec(t *testing.T) {
 							Name: buildah,
 						},
 					},
-				},
-				buildStrategy: &buildv1alpha1.BuildStrategy{
+				}
+
+				buildStrategy = &buildv1alpha1.BuildStrategy{
```

Comment:
@qu1queee I am getting error if I remove this line. Not so sure what you mean as redundant line.
From my understanding, we define the `BuildSteps: []buildv1alpha1.BuildStep` on top level, the ele inside should be the type of `buildv1alpha1.BuildStep`.

When I remove the line, the `Container: corev1.Container` would cause the error as the type should be `buildv1alpha1.BuildStep` not the `Container` in `[]buildv1alpha1.BuildStep`

For your reference :
```
				buildStrategy = &buildv1alpha1.BuildStrategy{
					ObjectMeta: metav1.ObjectMeta{Name: buildah},
					Spec: buildv1alpha1.BuildStrategySpec{
						BuildSteps: []buildv1alpha1.BuildStep{
							buildv1alpha1.BuildStep{
								Container: corev1.Container{
									Name:       "build",
									Image:      "$(build.builder.image)",
									WorkingDir: "/workspace/source",
									Command: []string{
										"buildah", "bud", "--tls-verify=false", "--layers", "-f", "$(build.dockerfile)", "-t", "$(build.output.image)", "$(build.source.contextDir)",
									},
									Args: []string{
										"buildah", "bud", "--tls-verify=false", "--layers", "-f", "$(build.dockerfile)", "-t", "$(build.output.image)", "$(build.source.contextDir)",
									},
									VolumeMounts: []corev1.VolumeMount{
										{
											Name:      "varlibcontainers",
											MountPath: "/var/lib/containers",
										},
									},
									SecurityContext: &corev1.SecurityContext{
										Privileged: &truePtr,
									},
								},
							},
							buildv1alpha1.BuildStep{
								Container: corev1.Container{
									VolumeMounts: []corev1.VolumeMount{
										{
											Name:      "varlibcontainers",
											MountPath: "/var/lib/containers",
										},
										{
											Name:      "something-else",
											MountPath: "/var/lib/containers",
										},
									},
								},
							},
```

Reply:
yes, you need to nest the whole `Container: corev1.Container{` , like:
```
BuildSteps: []buildv1alpha1.BuildStep{
							buildv1alpha1.BuildStep{
{
Container: corev1.Container{
...
```